infrastructure/modules/cdn-frontdoor-endpoint/README.md

```diff
@@ -0,0 +1,224 @@
+  origins = {
+    for region in keys(var.regions) : module.linux_web_app["${each.value.origin.webapp_key}-${region}"].name => merge(
+      each.value.origin,
+      {
+        hostname           = "${module.linux_web_app["${each.value.origin.webapp_key}-${region}"].name}.azurewebsites.net"
+        origin_host_header = "${module.linux_web_app["${each.value.origin.webapp_key}-${region}"].name}.azurewebsites.net"
+
+        private_link = var.features.private_endpoints_enabled ? {
+          target_type            = "sites" # "sites" for App Services, "managedEnvironments" for Container Apps.
+          location               = region
+          private_link_target_id = module.linux_web_app["${each.value.origin.webapp_key}-${region}"].id
+        } : null
+      }
+    )
+  }
+  public_dns_zone_rg_name = data.terraform_remote_state.hub.outputs.public_dns_zone_rg_name
+  resource_group_name     = data.terraform_remote_state.hub.outputs.project_rg_names["${var.application_full_name}-${local.primary_region}"]
+  route                   = each.value.route
+  security_policies       = each.value.security_policies
+
+  tags = var.tags
+}
+
+```
+
+**variables.tf** in root module
+```hcl
+variable "frontdoor_endpoint" {
+  description = "Configuration for Front Door"
+  type = map(object({
+    origin = object({
+      enabled                        = optional(bool, true)
+      http_port                      = optional(number, 80)  # 1–65535
+      https_port                     = optional(number, 443) # 1–65535
+      priority                       = optional(number, 1)   # 1–5
+      webapp_key                     = string                # From var.linux_web_app.linux_web_app_config
+      weight                         = optional(number, 500) # 1–1000
+    })
+
+    origin_group = optional(object({
+      health_probe = optional(object({
+        interval_in_seconds = number # Required: 1–255
+        path                = optional(string, "/")
+        protocol            = optional(string, "Https")
+        request_type        = optional(string, "HEAD")
+      }))
+
+      load_balancing = optional(object({
+        additional_latency_in_milliseconds = optional(number, 50) # Optional: 0–1000
+        sample_size                        = optional(number, 4)  # Optional: 0–255
+        successful_samples_required        = optional(number, 3)  # Optional: 0–255
+      }), {})
+
+      session_affinity_enabled                                  = optional(bool, true)
+      restore_traffic_time_to_healed_or_new_endpoint_in_minutes = optional(number)
+    }), {})
+
+    custom_domains = optional(map(object({
+      dns_zone_name = string
+      host_name     = string
+
+      tls = optional(object({
+        certificate_type         = optional(string, "ManagedCertificate")
+        cdn_frontdoor_secret_key = optional(string, null) # From var.projects[].frontdoor_profile.secrets in Hub
+      }), {})
+
+      zone_rg_name = optional(string, null)
```

Comment:
Is this useful? I haven't needed it in my review

Reply:
It adds 1 line for each zone yes. But it removes 1 variable and the logic with the `coalesce`. This makes it easier to read for the next engineer